tests/scattering2d/test_torch_scattering2d.py

```diff
@@ -327,6 +320,14 @@ def test_fft(self, backend):
         assert z.shape == x.shape[:-1]
         assert torch.allclose(y[..., 0], z)
 
+        x_r = x[..., 0].contiguous()
+        y[..., 1] = 0
+
+        z = backend.fft(x_r, 'R2C')
+
+        assert torch.allclose(y, z)
```

Comment:
Why should this be true? Taking the real part of `x` shouldn't result in its FT being real.

Reply:
lemme look into this

Reply:
![image](https://user-images.githubusercontent.com/39755015/84211222-b3a13600-aa6f-11ea-9600-3beb39ce4c96.png)

Using the unnormalized form of the 2d discrete Fourier transform (above is normalized), the way we calculate Y is correct. However, this example is kind of bad, since if we used some other number than M = N = 2, then we would have to worry about complex numbers coming into play, so my method of zeroing out the array works here, but in general it should not work. 

So we should use a better example here.